ratis-metrics/src/main/java/org/apache/ratis/metrics/MetricsReporting.java

```diff
@@ -34,17 +34,41 @@ private MetricsReporting() {
   }
 
   public static Consumer<RatisMetricRegistry> consoleReporter(TimeDuration rate) {
-    return ratisMetricRegistry -> ConsoleReporter.forRegistry(ratisMetricRegistry.getDropWizardMetricRegistry())
-        .convertRatesTo(TimeUnit.SECONDS).convertDurationsTo(TimeUnit.MILLISECONDS).build()
-        .start(rate.getDuration(), rate.getUnit());
+    return ratisMetricRegistry -> {
+      ConsoleReporter reporter = ConsoleReporter.forRegistry(ratisMetricRegistry.getDropWizardMetricRegistry())
+          .convertRatesTo(TimeUnit.SECONDS).convertDurationsTo(TimeUnit.MILLISECONDS).build();
+      reporter.start(rate.getDuration(), rate.getUnit());
+      ratisMetricRegistry.setConsoleReporter(reporter);
```

Comment:
Can you tell me why we need this change? Instead of returning a function that starts the console reporter for a given RatisMetricRegistry, why are we starting and using a setter?
      `ratisMetricRegistry.setConsoleReporter(reporter);`

Reply:
@avijayanhwx Because when we stop the reporter, we must get it from registry, so we need to set it in registry when create. Such as the following code, we need to getConsoleReporter from registry.
```
return ratisMetricRegistry -> {
      ConsoleReporter reporter = ratisMetricRegistry.getConsoleReporter();
      if (reporter != null) {
        reporter.close();
      }
    };
```